cmd/manager/main.go

```diff
@@ -49,52 +50,76 @@ var (
 )
 
 func init() {
-	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	if err := clientgoscheme.AddToScheme(scheme); err != nil {
+		panic(err)
+	}
 
-	podNamespace, ok := os.LookupEnv("POD_NAMESPACE")
-	if !ok {
-		panic("POD_NAMESPACE is not set")
+	//+kubebuilder:scaffold:scheme
+}
+
+func main() {
+	if err := cmd().Execute(); err != nil {
+		setupLog.Error(err, "error running manager")
+		os.Exit(1)
 	}
-	namespace = podNamespace
+}
 
-	k, ok := os.LookupEnv("NGROK_API_KEY")
-	if !ok {
-		panic("NGROK_API_KEY is not set")
+type managerOpts struct {
+	// flags
+	metricsAddr          string
+	enableLeaderElection bool
+	probeAddr            string
+	zapOpts              *zap.Options
+
+	// env vars
+	podNamespace string
+	ngrokAPIKey  string
+}
+
+func cmd() *cobra.Command {
+	var opts managerOpts
+	c := &cobra.Command{
+		Use: "manager",
+		RunE: func(c *cobra.Command, args []string) error {
+			return runController(c.Context(), opts)
+		},
 	}
-	ngrokApiKey = k
 
-	//+kubebuilder:scaffold:scheme
+	c.Flags().StringVar(&opts.metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to")
+	c.Flags().StringVar(&opts.probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	c.Flags().BoolVar(&opts.enableLeaderElection, "leader-elect", false, "Enable leader election for controller manager. Enabling this will ensure there is only one active controller manager.")
```

Comment:
Not relevant to your PR, but @mschenck maybe we should just remove this? There shouldn't be a case when you can disable leader election, things would break i believe. I'll make a follow up note